### PR TITLE
update to contracts v2.0.1

### PR DESCRIPTION
**Pull request by volodymyr-basiuk**

### CODE REVIEW

File: contracts/examples/ERC20Verifier.sol [hunk 1]
```diff
@@ -1,29 +1,39 @@
...
     mapping(address => uint256) public addressToId;
 
-    uint256 public TOKEN_AMOUNT_FOR_AIRDROP_PER_ID = 5 * 10**uint256(decimals());
+    uint256 public TOKEN_AMOUNT_FOR_AIRDROP_PER_ID;
 
-    constructor(string memory name_, string memory symbol_) ERC20(name_, symbol_) {}
+    function initialize(
+        string memory name,
+        string memory symbol,
+        address initialOwner
```
Comment (AndriianChestnykh) [inline contracts/examples/ERC20Verifier.sol:21]: I think we can omit this parameter for simplification and use `_msgSender()` for the `__ZKPVerifier_init()` method instead.

File: contracts/examples/ERC20Verifier.sol [hunk 2]
```diff
@@ -44,15 +57,21 @@ contract ERC20Verifier is ERC20, ZKPVerifier {
...
+        address from /* from */,
         address to,
-        uint256 /* amount */
-    ) internal view override {
+        uint256 amount /* amount */
+    ) internal override beforeTransfer(to) {
+        super._update(from, to, amount);
+    }
+
+    modifier beforeTransfer(address to) {
```
Comment (AndriianChestnykh) [inline contracts/examples/ERC20Verifier.sol:68]: put the modifier before all the methods according to Solidity code conventions

File: scripts/deployERC20SelectiveDisclosure.ts [hunk 3]
```diff
@@ -29,8 +29,12 @@ async function main() {
   const contractName = 'ERC20SelectiveDisclosureVerifier';
   const name = 'ERC20SelectiveDisclosureVerifier';
   const symbol = 'ERCZKP';
+  const owner = (await ethers.getSigners())[0];
   const ERC20ContractFactory = await ethers.getContractFactory(contractName);
-  const erc20instance = await ERC20ContractFactory.deploy(name, symbol);
+  const erc20instance = await upgrades.deployProxy(
+    ERC20ContractFactory,
+    [name, symbol, owner.address]
```
Comment (AndriianChestnykh) [inline scripts/deployERC20SelectiveDisclosure.ts:36]: The same here, use _msgSender() instead of the parameter.

File: test/utils/deploy-utils.ts [hunk 4]
```diff
@@ -65,8 +64,14 @@ export async function deployERC20ZKPVerifierToken(
...
   const ERC20Verifier = await ethers.getContractFactory('ERC20Verifier');
-  const erc20Verifier = await ERC20Verifier.deploy(name, symbol);
+  const signers = await ethers.getSigners();
+  const erc20Verifier = await upgrades.deployProxy(
+    ERC20Verifier,
+    [name, symbol, signers[0].address]
+  );
   await erc20Verifier.deployed();
+  // const erc20Verifier = await ERC20Verifier.deploy(name, symbol);
+  // await erc20Verifier.deployed();
```
Comment (AndriianChestnykh) [inline test/utils/deploy-utils.ts:74]: Remove the comment

File: contracts/examples/ERC20SelectiveDisclosureVerifier.sol [hunk 5]
```diff
@@ -45,20 +55,30 @@ contract ERC20SelectiveDisclosureVerifier is ERC20, ZKPVerifier {
...
+        _;
     }
 
     function getOperatorOutput() public view returns (uint256) {
         uint256 id = addressToId[_msgSender()];
-        require(id != 0, 'sender id doesn''t found');
+        require(
+            id != 0,
+            'sender id doesn'
+            't found'
```
Comment (AndriianChestnykh) [inline contracts/examples/ERC20SelectiveDisclosureVerifier.sol:80]: Strange formatting :)

File: contracts/examples/ERC20Verifier.sol (hunk 1 above)
Comment (volodymyr-basiuk) [reply contracts/examples/ERC20Verifier.sol]: removed

File: test/utils/deploy-utils.ts (hunk 4 above)
Comment (volodymyr-basiuk) [reply test/utils/deploy-utils.ts]: done

File: contracts/examples/ERC20SelectiveDisclosureVerifier.sol (hunk 5 above)
Comment (volodymyr-basiuk) [reply contracts/examples/ERC20SelectiveDisclosureVerifier.sol]: fixed, linter

**Pull request closed without merging**